testing/s3mock/src/main/java/org/projectnessie/s3mock/data/Bucket.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.s3mock.data;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableBucket.class)
+@JsonDeserialize(as = ImmutableBucket.class)
+public interface Bucket {
```

Comment:
Can we just depend on this library (https://github.com/adobe/S3Mock), instead of we implementing or maintaining the similar/same code?

Reply:
If you look closely at both implementations, you see that those do _different_ things.

Reply:
I do find the data folder in this PR to be similar to https://github.com/adobe/S3Mock/tree/main/server/src/main/java/com/adobe/testing/s3mock/dto

But yeah, I do have to look closely as you mentioned.

Reply:
imo this is a legitimate question, so my suggestion would be to briefly summarize the reasoning on why we can not use this library in a README file inside the module folder, WDYT ?
this way we dont need to expect everyone to look at both implementations closely to tell what the exact difference is and why we had to roll our own.

Reply:
Again - the Adobe S3Mock does something very different. This one does intentionally _NOT_ serve "real" content.

Reply:
Not sure why comparing with another library, that serves a different purpose, would make sense.
This one serves content and listings from _functions_ not a real file system incl put operations.

Reply:
Thanks for the explanation. I may need a while to understand all these object store interface implementations as I am not very familiar with them. 

If no one else reviews, I will dig deep to understand these.